### PR TITLE
Support write operation of user label list for User Label Cluster

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* No write support for User Label cluster  

#### Change overview
Support write operation of user label list for User Label cluster

#### Testing
How was this tested? (at least one bullet point required)
* No write list attribute support in chip-tool yet, the test is done by auto generated test case in CHIPClustersTests.m

### CODE REVIEW

File: src/app/clusters/user-label-server/user-label-server.cpp [hunk 1]
```diff
@@ -74,7 +78,27 @@ CHIP_ERROR UserLabelAttrAccess::ReadLabelList(EndpointId endpoint, AttributeValu
...
+    auto iter     = decodablelist.begin();
+    while (iter.Next())
+    {
+        auto & entry = iter.GetValue();
+
+        VerifyOrReturnError(index < DeviceLayer::kMaxUserLabels, CHIP_ERROR_BUFFER_TOO_SMALL);
+        labelList.add(entry);
+    }
+
+    ReturnErrorOnFailure(DeviceLayer::PlatformMgr().SetUserLabelList(endpoint, labelList));
```
Comment (bzbarsky-apple) [inline src/app/clusters/user-label-server/user-label-server.cpp:98]: Why are we doing this before checking `iter.GetStatus()`?  If the incoming value is not actually decodable as a valid list, do we want to overwrite the stored value with an empty list?  This does not seem right.

File: src/app/clusters/user-label-server/user-label-server.cpp [hunk 2]
```diff
@@ -74,7 +78,27 @@ CHIP_ERROR UserLabelAttrAccess::ReadLabelList(EndpointId endpoint, AttributeValu
     return err;
 }
 
-UserLabelAttrAccess gAttrAccess;
+CHIP_ERROR UserLabelAttrAccess::WriteLabelList(EndpointId endpoint, AttributeValueDecoder & aDecoder)
+{
+    DeviceLayer::LabelList<app::Clusters::UserLabel::Structs::LabelStruct::Type, DeviceLayer::kMaxUserLabels> labelList;
```
Comment (bzbarsky-apple) [inline src/app/clusters/user-label-server/user-label-server.cpp:83]:
```suggestion
    DeviceLayer::LabelList<Structs::LabelStruct::Type, DeviceLayer::kMaxUserLabels> labelList;
```

File: src/app/clusters/user-label-server/user-label-server.cpp [hunk 3]
```diff
@@ -74,7 +78,27 @@ CHIP_ERROR UserLabelAttrAccess::ReadLabelList(EndpointId endpoint, AttributeValu
...
+
+    ReturnErrorOnFailure(aDecoder.Decode(decodablelist));
+
+    uint8_t index = 0;
+    auto iter     = decodablelist.begin();
+    while (iter.Next())
+    {
+        auto & entry = iter.GetValue();
+
+        VerifyOrReturnError(index < DeviceLayer::kMaxUserLabels, CHIP_ERROR_BUFFER_TOO_SMALL);
```
Comment (bzbarsky-apple) [inline src/app/clusters/user-label-server/user-label-server.cpp:94]: Why do we need a separate `index`?  Why can't we use `labelList.size()`?   Or just use the fact that `add()` returns error on out of space and is already doing this length check anyway?

Comment (yufengwangca) [reply src/app/clusters/user-label-server/user-label-server.cpp]: Acked

**Review by yunhanw-google — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**